test/domain/contacts/contacts_manager_test.dart

```diff
@@ -0,0 +1,462 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'contacts_manager_test.mocks.dart';
+
+@GenerateMocks([GetTomContactsInteractor, PhonebookContactInteractor])
+void main() {
+  final List<Contact> tomContacts = [
+    const Contact(
+      email: "alice1@domain.com",
+      displayName: "Alice 1",
+      matrixId: "@alice1:domain.com",
+      phoneNumber: "07 81 12 38 61",
+      status: ContactStatus.active,
+    ),
+    const Contact(
+      email: "alice2@domain.com",
+      displayName: "Alice 2",
+      matrixId: "@alice2:domain.com",
+      phoneNumber: "02 81 12 38 61",
+      status: ContactStatus.active,
+    ),
+    const Contact(
+      email: "alice3@domain.com",
+      displayName: "Alice 3",
+      matrixId: "@alice3:domain.com",
+      phoneNumber: "03 81 12 38 61",
+      status: ContactStatus.active,
+    ),
+    const Contact(
+      email: "alice4@domain.com",
+      displayName: "Alice 4",
+      matrixId: "@alice4:domain.com",
+      phoneNumber: "04 81 12 38 61",
+      status: ContactStatus.inactive,
+    ),
+    const Contact(
+      email: "alice5@domain.com",
+      displayName: "Alice 5",
+      matrixId: "@alice5:domain.com",
+      phoneNumber: "05 81 12 38 61",
+      status: ContactStatus.inactive,
+    ),
+  ];
+
+  final List<Contact> phonebookContacts = [
+    const Contact(
+      email: "bob@domain.com",
+      displayName: "BoB",
+    ),
+    const Contact(
+      displayName: "BoB1",
+      phoneNumber: "11 22 33 44 55",
+    ),
+    const Contact(
+      email: "bob2@domain.com",
+      displayName: "BoB2",
+      phoneNumber: "+84000000000",
+    ),
+    const Contact(
+      email: "bob3@domain.com",
+      displayName: "BoB 3",
+    ),
+    const Contact(
+      email: "bob@domain.com",
+      displayName: "BoB",
+    ),
+  ];
+
+  setUpAll(() {
+    HiveCacheStore('${Directory.current.path}/test/data/file_store');
+    TestWidgetsFlutterBinding.ensureInitialized();
+    GetItInitializer().setUp();
+  });
+
+  group('Request permission on web and tomContacts', () {
+    late ContactsManager contactsManager;
+    late MockGetTomContactsInteractor mockGetTomContactsInteractor;
+
+    setUp(() {
+      getIt.registerSingleton<MockGetTomContactsInteractor>(
+        MockGetTomContactsInteractor(),
+      );
+    });
+
+    test('TomContact is Empty', () async {
+      contactsManager = getIt.get<ContactsManager>();
+
+      mockGetTomContactsInteractor = getIt.get<MockGetTomContactsInteractor>();
+
+      when(
+        mockGetTomContactsInteractor.execute(
+          limit: AppConfig.maxFetchContacts,
+        ),
+      ).thenAnswer(
+        (_) async* {
+          yield const Right(ContactsLoading());
+          yield const Right(GetContactsSuccess(contacts: []));
+        },
+      );
+
+      contactsManager.initialSynchronizeContacts(
+        isAvailableSupportPhonebookContacts: false,
+      );
```

Comment:
that is the function we want to test, we will verify the outputs of it and also the behavior